src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int NUMBER_OF_ELEMENTS = 3;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        for (String item: inputReport) {
+            String[] arrayFromItem = item.replaceAll(" ", "").split(",");
+            if (arrayFromItem.length > NUMBER_OF_ELEMENTS) {
+                throw new RuntimeException("The number of elements in a file line"
+                        + " is greater than 3");
```

Comment:
The error message could be more informative by including the actual line content or index that caused the error. This would help in debugging and understanding which line in the input is problematic.